vm/src/obj/objtuple.rs

```diff
@@ -234,6 +235,10 @@ impl PyTuple {
         vm: &VirtualMachine,
     ) -> PyResult<PyTupleRef> {
         let elements = if let OptionalArg::Present(iterable) = iterable {
+            if cls.is(&vm.ctx.tuple_type()) && iterable.lease_class().is(&vm.ctx.tuple_type()) {
```

Comment:
Should both class be tuple type? I guess we just need to check `iterable.lease_class().is(cls)` because we can reuse the value when the type is same.